docs/protocol/specifications/vm1.md

```diff
@@ -1121,7 +1121,9 @@ Code  | Stack Diagram                                        | Cost
     2. If the entry satisfies all of the following conditions pushes [true](#vm-boolean) on the data stack; otherwise pushes [false](#vm-boolean):
         1. the destination entry is an [output](blockchain.md#output-1) or a [retirement](blockchain.md#retirement-1),
         2. if the destination is an output: control program equals `prog` and VM version equals `version`,
-        3. if the destination is a retirement: `prog` is an empty string and `version` is zero,
+        3. if the destination is a retirement:
+            * `version` must be zero,
+            * if [expansion flag](#vm-sate) is `false`, `prog` must be an empty string.
```

Comment:
So the risk here is:

Core was updated with new txhash computation via entries, but for retirement-using transactions SDK/API was still creating signature programs `CHECKOUTPUT(vm=1,prog="FAIL*")`. If we update validation logic to require `CHECKOUTPUT(vm=0,prog="")` we'd make these checks fail which means we'd need a blockchain reset.

But if we allow `CHECKOUTPUT(vm=1,prog="FAIL*")` for compatibility, we'll introduce a malleability — someone can satisfy that requirement with both normal retirement and ugly retirement (that clobbers UTXO set). Which may break some application-level accounting logic. To mitigate that we can soft-fork a rule that forbids FAIL-prefixed outputs outright so people have to use proper retirements.